algosdk/v2client/indexer.py

```diff
@@ -926,6 +926,44 @@ def application_logs(
 
         return self.indexer_request("GET", req, query, **kwargs)
 
+    def application_box_by_name(
+        self, application_id: int, box_name: bytes, **kwargs
+    ):
+        """
+        Return the value of an application's box.
+
+        Args:
+            application_id (int): application index
+            box_name (bytes): The name (key) of the box.
+        """
+        encoded_box = base64.b64encode(box_name).decode()
+        box_name_encoded = "b64:" + encoded_box
+        req = "/applications/" + str(application_id) + "/box"
+        params = {"name": box_name_encoded}
+
+        return self.indexer_request("GET", req, params, **kwargs)
+
+    def application_boxes(
+        self, application_id: int, limit: int = 0, next_page=None, **kwargs
+    ):
+        """
+        Return a list of all the application's boxes.
+
+        Args:
+            application_id (int): The ID of the application to look up.
+            limit (int, optional): Max number of box names to return.
+                If max is not set, or max == 0, returns all box-names.
+            next_page (string, optional): used for pagination
```

Comment:
`next_page` is a box name and expects the same encoding methodology as the `box_name` parameter of `application_box_by_name()`. As an example, [consider this](https://github.com/algorand/indexer/blob/590251145af5b0d08d37bed911caacc5b08681be/api/test_resources/boxes.json#L566-L605).

We should be clear about how this parameter should be passed back. I see two ways going forward:
1. callers provide `next_page` already `b64` encoded. This makes sense because this allows them to plug in _without any processing_ the value of `next-token` which they received from the previous call.
2. alternatively, they could provide the box name in the same format as in the call to `application_box_by_name()`. The downside is that the user whould first have to b64 _decode_ and then provide, and then this method would encoded it again. 

I favor appoach (1) because it's more efficient. But I could see arguments for (2).

In either case, the prefix `b64:` needs to be pre-prended to `next_page`, i.e. `f"b64:{next_page}"` , before gelling **GET**.

Reply:
That is a good point. On the perspective of a caller, when we are querying for `next_page`, we would wish for passing in a string (or `Uint8Array`) that is close to readable format.

Another concern is that, if we want to prepend  `b64:` to encoded `next_page`, we may want to check if `next_page` is empty or not, or as you suggested in js-sdk, the resulting query would be funny with only `b64:`.

Reply:
Going to take back what I thought about `b64:` empty string, if that is valid in indexer query, then we should be fine with it.

Reply:
Yes, it's ok (if a bit strange) to give empty `b64:` Indexer will just query starting at this minimum of all strings and return the same results as if not provided `b64:`... ideally, it's better not to provide if it's not too big a hassle.

Reply:
just circled back from langspec after a while, and I am in favor of option 1, which is caller provides `next_page` to be `b64` encoded.

My additional thought is that, but the method should check if `next_page` is `None` or `""` before prefixing with `b64:`, though I assume it is find for indexer to accept things like `b64:` with empty string.

Standing on the perspective of debugging, or making queries more 'cleaner', if one do not want to add `next_page`, then it should not appear in query tho... and I think the change across SDKs should be low effort.

Reply:
From the decision we made in https://github.com/algorand/algorand-sdk-testing/pull/219, we keep the `nextToken` method unchanged.